src/ucs/stats/stats.c

```diff
@@ -602,13 +619,12 @@ static size_t ucs_stats_aggregate_sum_recurs(ucs_stats_node_t *node,
     ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
         ucs_list_for_each(temp_node, &filter_node->type_list_head, type_list) {
             cnt_agrgt_sum_index = counter_output_index +
-                                  ucs_stats_clsid_to_agrgt_sum_base_offset_get(
+                                  ucs_stats_clsid_to_base_offset(
                                           node, num_counters_filtered);
 
-            if (ucs_likely(cnt_agrgt_sum_index < num_stats_counters_max)) {
-                aggregate_sum[cnt_agrgt_sum_index] +=
-                        temp_node->counters[counter_index];
-            }
+            ucs_assert(cnt_agrgt_sum_index < num_stats_counters_max);
```

Comment:
it's not ok: the user passes some max value and expects no more counters will be returned. we should respect this parameter and not crash the program if more counters exists. it's as if snprintf() will crash if the resulting string is >max.

Reply:
Fixed.